All files updated to work with Hubspot API

BeAllAround opened the pull request:
Sorry if there are too many commits and sorry I didn't modify the files directly from the repository so that the code is visible in the "changes" history - at least on GitHub. I found it easier to delete them and upload new ones.

I am sure you know everything I am about to say but I am just making sure we are on the same page.

Run "$ npm install" and provide the file "local-config.js" in the test directory. After that, just set accessToken and $HUBSPOT_ACCESS_TOKEN to the private token I picked from the website. It all runs pretty as expected. Furthermore, I commented out the 'role: entity' actions in the file "live.js." - which also run along with the other tests in the file.

I am looking forward to your reply.

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,103 @@
+							  let id = data[0];
+							  let obj = {};
+							  data = data.slice(1);
+							  try{
```
no need for try catch here - the function is async so the exception will be caught further up the call stack

BeAllAround commented:
I committed all of your changes!

Regards!

On Mon, Jun 27, 2022 at 11:04 AM Richard Rodger ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907132219>
> :
>
> > @@ -30,33 +30,33 @@ that they can be accessed using the Seneca entity API and messages.
>  Seneca()
>    .use('provider', {
>      provider: {
> -      trello: {
> +      hubspot: {
>
> update this to show usage with the env plugin as per test/live.js
> ------------------------------
>
> In README.md
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907133103>
> :
>
> >
> -let repo = await seneca.entity('provider/trello/repo')
> -  .load$('senecajs/trello-api-test')
> +let company = await seneca.entity('provider/hubspot/company')
> +  .load$('id/type/name/city/state/description')
>
> this should just be a company identifier
> ------------------------------
>
> In README.md
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907133876>
> :
>
> > @@ -73,7 +73,7 @@ Set plugin options when loading with:
>  ```js
>
>
> -seneca.use('TrelloProvider', { name: value, ... })
> +seneca.use('HubspotProvider', { name: value, ... })
>
> should be hubspot-provider
> ------------------------------
>
> In package.json
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907135554>
> :
>
> >    "keywords": [
>      "seneca",
> -    "trello",
> -    "trello-provider"
> +    "hubspot",
> +    "hubspot-provider"
>    ],
>    "author": "Richard Rodger - richardrodger.com",
>
> Add yourself as a contributor :)
>
> https://docs.npmjs.com/cli/v8/configuring-npm/package-json#people-fields-author-contributors
> ------------------------------
>
> In package.json
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907136134>
> :
>
> >    ],
>    "author": "Richard Rodger - richardrodger.com",
>    "repository": {
>      "type": "git",
> -    "url": "git://trello.com/senecajs/seneca-trello-provider.git"
> +    "url": "git://hubspot.com/senecajs/seneca-hubspot-provider.git"
>
> an error in the original - this should be github.com
> ------------------------------
>
> In package.json
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907137205>
> :
>
> >      "seneca-msg-test": "^3.4.0",
> +    "ts-jest": "^28.0.5",
>
> this should not be necessary as we use use esbuild for speed
> ------------------------------
>
> In src/hubspot-provider.d.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907138067>
> :
>
> > @@ -0,0 +1,7 @@
> +declare type HubspotProviderOptions = {};
>
> this should not be needed as it is automatically created
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907139538>
> :
>
> > @@ -0,0 +1,103 @@
> +const Pkg = require('../package.json');
> +const hubspot = ***@***.***/api-client');
> +
> +type HubspotProviderOptions = {}
> +
> +function HubspotProvider(this: any, options : HubspotProviderOptions) {
> +	const seneca:any = this;
>
> seneca indent convention is 2 spaces - run prettier to get this
> also we don't use semis
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907140683>
> :
>
> > @@ -0,0 +1,103 @@
> +const Pkg = require('../package.json');
> +const hubspot = ***@***.***/api-client');
>
> capitalize - this is also a convention in Seneca code
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907142822>
> :
>
> > +			sdk: {
> +				name: 'hubspot-node',
> +				version: Pkg.dependencies['hubspot'],
> +			}
> +		}
> +	}
> +	entityBuilder(this, {
> +		  provider: {
> +			  name: 'hubspot'
> +		  },
> +		  entity: {
> +			  company: {
> +				  cmd: {
> +					  list: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let res, list;
>
> unless you are specifying types, no need to predefine vars
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907143681>
> :
>
> > +				name: 'hubspot-node',
> +				version: Pkg.dependencies['hubspot'],
> +			}
> +		}
> +	}
> +	entityBuilder(this, {
> +		  provider: {
> +			  name: 'hubspot'
> +		  },
> +		  entity: {
> +			  company: {
> +				  cmd: {
> +					  list: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let res, list;
> +							  res = (await this.shared.sdk.crm.companies.basicApi.getPage(10)).results;
>
> const res = ...
> let list = ...
>
> avoid hard-coded values in code: getPage(10)
> this value should come from msg.q.limit$ see
> https://senecajs.org/docs/tutorials/understanding-query-syntax.html#wp-limit
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907146253>
> :
>
> > +		  },
> +		  entity: {
> +			  company: {
> +				  cmd: {
> +					  list: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let res, list;
> +							  res = (await this.shared.sdk.crm.companies.basicApi.getPage(10)).results;
> +							  list = res.map((data) => entize(data))
> +							  return list;
> +						  }
> +					  },
> +					  load: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let data = (msg.q.id.split('/'));
> +							  let id = data[0];
>
> what does hubspot use as the unique id from companies? that should be
> sufficient
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907147430>
> :
>
> > +				  cmd: {
> +					  list: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let res, list;
> +							  res = (await this.shared.sdk.crm.companies.basicApi.getPage(10)).results;
> +							  list = res.map((data) => entize(data))
> +							  return list;
> +						  }
> +					  },
> +					  load: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let data = (msg.q.id.split('/'));
> +							  let id = data[0];
> +							  let obj = {};
> +							  data = data.slice(1);
> +							  try{
>
> no need for try catch here - the function is async so the exception will
> be caught further up the call stack
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907148450>
> :
>
> > +			  name: 'hubspot'
> +		  },
> +		  entity: {
> +			  company: {
> +				  cmd: {
> +					  list: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let res, list;
> +							  res = (await this.shared.sdk.crm.companies.basicApi.getPage(10)).results;
> +							  list = res.map((data) => entize(data))
> +							  return list;
> +						  }
> +					  },
> +					  load: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let data = (msg.q.id.split('/'));
>
> unless the value really is generic "data" avoid using data as a variable
> name
> in this case idParts might be better
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907149551>
> :
>
> > +					  list: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let res, list;
> +							  res = (await this.shared.sdk.crm.companies.basicApi.getPage(10)).results;
> +							  list = res.map((data) => entize(data))
> +							  return list;
> +						  }
> +					  },
> +					  load: {
> +						  action: async function(this: any, entize: any, msg: any) {
> +							  let data = (msg.q.id.split('/'));
> +							  let id = data[0];
> +							  let obj = {};
> +							  data = data.slice(1);
> +							  try{
> +								  obj = await this.shared.sdk.crm.companies.basicApi.getById(id, data.length != 0 ? data : undefined);
>
> what a strange API! - please add a comment with a link to the hubspot docs
> for this method
> ------------------------------
>
> In test/basic.messages.js
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r907150553>
> :
>
> >          version: Pkg.version,
>          sdk: {
> -          name: 'trello-node',
> -          version: Pkg.dependencies['trello'],
> +          name: 'hubspot-node',
>
> the package name is just hubspot
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#pullrequestreview-1019828277>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJILHFXXOUXJG5SWDRUFGT3VRFVDDANCNFSM5Z4TKO5Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

rjrodger commented on README.md:
```diff
@@ -30,33 +30,33 @@ that they can be accessed using the Seneca entity API and messages.
-let repo = await seneca.entity('provider/trello/repo')
-  .load$('senecajs/trello-api-test')
+let company = await seneca.entity('provider/hubspot/company')
+  .load$('id/type/name/city/state/description')
```
TODO @rjrodger explain Seneca fields$ directive

rjrodger replied:
@BeAllAround see https://github.com/senecajs/seneca-entity/blob/master/test/entity.test.js#L377 for the "seneca way" to do this

the .load$ method should only accept either a single value as id, or a query object.

return all the standard hubspot fields by default, or parse the fields$ directive to get the fields that the user wants

```
seneca.entity('provider/hubspot/company').load$('ID-VALUE')
seneca.entity('provider/hubspot/company').load$({id:'ID-VALUE'})
seneca.entity('provider/hubspot/company').load$({someField:'FIELD-VALUE'}) // same as SQL WHERE someField = "FIELD-VALUE"

seneca.entity('provider/hubspot/company').load$({id:'ID-VALUE', fields$:['type','name']}) // only load the type and name fields
```

BeAllAround commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,103 @@
+type HubspotProviderOptions = {}
+
+function HubspotProvider(this: any, options : HubspotProviderOptions) {
+	const seneca:any = this;
```
Done!

rjrodger commented on src/HubspotProvider-doc.d.ts:
```diff
@@ -0,0 +1,6 @@
+declare const docs: {
```
shouldn't this be in the dist folder?

rjrodger commented on src/HubspotProvider-doc.ts:
```diff
@@ -0,0 +1,12 @@
+const docs = {
+	get_info: {
```
indentation should be 2 spaces

rjrodger commented on src/hubspot-provider.d.ts:
```diff
@@ -0,0 +1,6 @@
+declare function HubspotProvider(this: any, options: HubspotProviderOptions): {
```
also, should this not be in dist?
or did you define this manually?

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,101 @@
+  const entityBuilder = this.export('provider/entityBuilder')
+
+  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
+  async function get_info(this: any, _msg: any) {
```
add a line to separate here
if more messages are added later, we want all the messages to be together
see: https://github.com/mgp/book-notes/blob/master/the-non-designers-design-book.markdown - proximity

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,101 @@
+      }
+    }
+  }
+ entityBuilder(this, {
```
same here - line separation needed

rjrodger replied:
also, the indentation is incorrect

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,101 @@
+       cmd: {
+         list: {
+	     action: async function(this: any, entize: any, msg: any) {
+	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
```
use `limit$`

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,101 @@
+	   action: async function(this: any, entize: any, msg: any) {
+	     let idParts = msg.q.fields$
+	     let id = msg.q.id
+	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
```
this line is too long - aim for max 80 chars (makes things like diff comparison easier)
place the comment above the line

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,101 @@
+	     let id = ent.id
+	     let desc = ent.properties.description
+	     let obj
+	     try{
```
missing a space - just run https://prettier.io and all these formatting problems go away!

rjrodger commented on src/hubspot-provider.ts:
```diff
@@ -0,0 +1,101 @@
+     }
+   }
+ })
+
```
too much space

in general, use 2 lines for "major" contextual differences

BeAllAround commented:
> @@ -0,0 +1,6 @@
+declare function HubspotProvider(this: any, options: HubspotProviderOptions): {

also, should this not be in dist?
or did you define this manually?


This was in the Trello provider like that.

On Mon, Jul 4, 2022, 1:44 PM Richard Rodger ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In src/HubspotProvider-doc.d.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912913788>
> :
>
> > @@ -0,0 +1,6 @@
> +declare const docs: {
>
> shouldn't this be in the dist folder?
> ------------------------------
>
> In src/HubspotProvider-doc.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912914601>
> :
>
> > @@ -0,0 +1,12 @@
> +const docs = {
> +	get_info: {
>
> indentation should be 2 spaces
> ------------------------------
>
> In src/hubspot-provider.d.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912915456>
> :
>
> > @@ -0,0 +1,6 @@
> +declare function HubspotProvider(this: any, options: HubspotProviderOptions): {
>
> also, should this not be in dist?
> or did you define this manually?
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917054>
> :
>
> > @@ -0,0 +1,101 @@
> +const Pkg = require('../package.json');
> +
> +const Hubspot = ***@***.***/api-client')
> +
> +
> +type HubspotProviderOptions = {}
> +
> +function HubspotProvider(this: any, options : HubspotProviderOptions) {
> +  const seneca:any = this
> +
> +  const entityBuilder = this.export('provider/entityBuilder')
> +
> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
> +  async function get_info(this: any, _msg: any) {
>
> add a line to separate here
> if more messages are added later, we want all the messages to be together
> see:
> https://github.com/mgp/book-notes/blob/master/the-non-designers-design-book.markdown
> - proximity
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917262>
> :
>
> > +
> +  const entityBuilder = this.export('provider/entityBuilder')
> +
> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
> +  async function get_info(this: any, _msg: any) {
> +    return {
> +      ok: true,
> +      name: 'hubspot',
> +      version: Pkg.version,
> +      sdk: {
> +        name: 'hubspot',
> +	version: Pkg.dependencies['hubspot'],
> +      }
> +    }
> +  }
> + entityBuilder(this, {
>
> same here - line separation needed
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912918048>
> :
>
> > +      sdk: {
> +        name: 'hubspot',
> +	version: Pkg.dependencies['hubspot'],
> +      }
> +    }
> +  }
> + entityBuilder(this, {
> +   provider: {
> +     name: 'hubspot'
> +   },
> +   entity: {
> +     company: {
> +       cmd: {
> +         list: {
> +	     action: async function(this: any, entize: any, msg: any) {
> +	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
>
> use limit$
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912918761>
> :
>
> > +     company: {
> +       cmd: {
> +         list: {
> +	     action: async function(this: any, entize: any, msg: any) {
> +	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
> +	       let res = (await this.shared.sdk.crm.companies.basicApi.getPage(limit)).results
> +	       let list = res.map((data) => entize(data))
> +	       return list
> +	     }
> +	 },
> +
> +	 load: {
> +	   action: async function(this: any, entize: any, msg: any) {
> +	     let idParts = msg.q.fields$
> +	     let id = msg.q.id
> +	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
>
> this line is too long - aim for max 80 chars (makes things like diff
> comparison easier)
> place the comment above the line
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912919528>
> :
>
> > +	 load: {
> +	   action: async function(this: any, entize: any, msg: any) {
> +	     let idParts = msg.q.fields$
> +	     let id = msg.q.id
> +	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
> +	     return entize(obj)
> +	   }
> +	 },
> +
> +	 save: {
> +	   action: async function(this: any, entize: any, msg: any) {
> +	     let ent = msg.ent
> +	     let id = ent.id
> +	     let desc = ent.properties.description
> +	     let obj
> +	     try{
>
> missing a space - just run https://prettier.io and all these formatting
> problems go away!
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912920034>
> :
>
> > +	       obj = await this.shared.sdk.crm.companies.basicApi.update(id, {properties: {description: desc}})
> +	     }
> +	     catch(err){
> +	       if(err.code >= 400 && err.code < 500){
> +		 return null
> +	       }
> +	       throw err
> +	     }
> +	     return entize(obj)
> +	   }
> +	 }
> +       }
> +     }
> +   }
> + })
> +
>
> too much space
>
> in general, use 2 lines for "major" contextual differences
> ------------------------------
>
> In src/hubspot-provider.ts
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917701>
> :
>
> > +
> +  const entityBuilder = this.export('provider/entityBuilder')
> +
> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
> +  async function get_info(this: any, _msg: any) {
> +    return {
> +      ok: true,
> +      name: 'hubspot',
> +      version: Pkg.version,
> +      sdk: {
> +        name: 'hubspot',
> +	version: Pkg.dependencies['hubspot'],
> +      }
> +    }
> +  }
> + entityBuilder(this, {
>
> also, the indentation is incorrect
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#pullrequestreview-1027424817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJILHFSNLQOAGO4KLYCORM3VSLFB7ANCNFSM5Z4TKO5Q>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

BeAllAround commented:
You are right actually. Sorry about that. It seems like jest generated it

On Mon, Jul 4, 2022, 1:58 PM Alex Milenkovic ***@***.***> wrote:

> > @@ -0,0 +1,6 @@
> +declare function HubspotProvider(this: any, options: HubspotProviderOptions): {
>
> also, should this not be in dist?
> or did you define this manually?
>
>
> This was in the Trello provider like that.
>
> On Mon, Jul 4, 2022, 1:44 PM Richard Rodger ***@***.***>
> wrote:
>
>> ***@***.**** requested changes on this pull request.
>> ------------------------------
>>
>> In src/HubspotProvider-doc.d.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912913788>
>> :
>>
>> > @@ -0,0 +1,6 @@
>> +declare const docs: {
>>
>> shouldn't this be in the dist folder?
>> ------------------------------
>>
>> In src/HubspotProvider-doc.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912914601>
>> :
>>
>> > @@ -0,0 +1,12 @@
>> +const docs = {
>> +	get_info: {
>>
>> indentation should be 2 spaces
>> ------------------------------
>>
>> In src/hubspot-provider.d.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912915456>
>> :
>>
>> > @@ -0,0 +1,6 @@
>> +declare function HubspotProvider(this: any, options: HubspotProviderOptions): {
>>
>> also, should this not be in dist?
>> or did you define this manually?
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917054>
>> :
>>
>> > @@ -0,0 +1,101 @@
>> +const Pkg = require('../package.json');
>> +
>> +const Hubspot = ***@***.***/api-client')
>> +
>> +
>> +type HubspotProviderOptions = {}
>> +
>> +function HubspotProvider(this: any, options : HubspotProviderOptions) {
>> +  const seneca:any = this
>> +
>> +  const entityBuilder = this.export('provider/entityBuilder')
>> +
>> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
>> +  async function get_info(this: any, _msg: any) {
>>
>> add a line to separate here
>> if more messages are added later, we want all the messages to be together
>> see:
>> https://github.com/mgp/book-notes/blob/master/the-non-designers-design-book.markdown
>> - proximity
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917262>
>> :
>>
>> > +
>> +  const entityBuilder = this.export('provider/entityBuilder')
>> +
>> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
>> +  async function get_info(this: any, _msg: any) {
>> +    return {
>> +      ok: true,
>> +      name: 'hubspot',
>> +      version: Pkg.version,
>> +      sdk: {
>> +        name: 'hubspot',
>> +	version: Pkg.dependencies['hubspot'],
>> +      }
>> +    }
>> +  }
>> + entityBuilder(this, {
>>
>> same here - line separation needed
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912918048>
>> :
>>
>> > +      sdk: {
>> +        name: 'hubspot',
>> +	version: Pkg.dependencies['hubspot'],
>> +      }
>> +    }
>> +  }
>> + entityBuilder(this, {
>> +   provider: {
>> +     name: 'hubspot'
>> +   },
>> +   entity: {
>> +     company: {
>> +       cmd: {
>> +         list: {
>> +	     action: async function(this: any, entize: any, msg: any) {
>> +	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
>>
>> use limit$
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912918761>
>> :
>>
>> > +     company: {
>> +       cmd: {
>> +         list: {
>> +	     action: async function(this: any, entize: any, msg: any) {
>> +	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
>> +	       let res = (await this.shared.sdk.crm.companies.basicApi.getPage(limit)).results
>> +	       let list = res.map((data) => entize(data))
>> +	       return list
>> +	     }
>> +	 },
>> +
>> +	 load: {
>> +	   action: async function(this: any, entize: any, msg: any) {
>> +	     let idParts = msg.q.fields$
>> +	     let id = msg.q.id
>> +	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
>>
>> this line is too long - aim for max 80 chars (makes things like diff
>> comparison easier)
>> place the comment above the line
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912919528>
>> :
>>
>> > +	 load: {
>> +	   action: async function(this: any, entize: any, msg: any) {
>> +	     let idParts = msg.q.fields$
>> +	     let id = msg.q.id
>> +	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
>> +	     return entize(obj)
>> +	   }
>> +	 },
>> +
>> +	 save: {
>> +	   action: async function(this: any, entize: any, msg: any) {
>> +	     let ent = msg.ent
>> +	     let id = ent.id
>> +	     let desc = ent.properties.description
>> +	     let obj
>> +	     try{
>>
>> missing a space - just run https://prettier.io and all these formatting
>> problems go away!
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912920034>
>> :
>>
>> > +	       obj = await this.shared.sdk.crm.companies.basicApi.update(id, {properties: {description: desc}})
>> +	     }
>> +	     catch(err){
>> +	       if(err.code >= 400 && err.code < 500){
>> +		 return null
>> +	       }
>> +	       throw err
>> +	     }
>> +	     return entize(obj)
>> +	   }
>> +	 }
>> +       }
>> +     }
>> +   }
>> + })
>> +
>>
>> too much space
>>
>> in general, use 2 lines for "major" contextual differences
>> ------------------------------
>>
>> In src/hubspot-provider.ts
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917701>
>> :
>>
>> > +
>> +  const entityBuilder = this.export('provider/entityBuilder')
>> +
>> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
>> +  async function get_info(this: any, _msg: any) {
>> +    return {
>> +      ok: true,
>> +      name: 'hubspot',
>> +      version: Pkg.version,
>> +      sdk: {
>> +        name: 'hubspot',
>> +	version: Pkg.dependencies['hubspot'],
>> +      }
>> +    }
>> +  }
>> + entityBuilder(this, {
>>
>> also, the indentation is incorrect
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#pullrequestreview-1027424817>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJILHFSNLQOAGO4KLYCORM3VSLFB7ANCNFSM5Z4TKO5Q>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

BeAllAround commented:
I think we are all set now!

On Mon, Jul 4, 2022 at 2:05 PM Alex Milenkovic ***@***.***> wrote:

> You are right actually. Sorry about that. It seems like jest generated it
>
> On Mon, Jul 4, 2022, 1:58 PM Alex Milenkovic ***@***.***> wrote:
>
>> > @@ -0,0 +1,6 @@
>> +declare function HubspotProvider(this: any, options: HubspotProviderOptions): {
>>
>> also, should this not be in dist?
>> or did you define this manually?
>>
>>
>> This was in the Trello provider like that.
>>
>> On Mon, Jul 4, 2022, 1:44 PM Richard Rodger ***@***.***>
>> wrote:
>>
>>> ***@***.**** requested changes on this pull request.
>>> ------------------------------
>>>
>>> In src/HubspotProvider-doc.d.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912913788>
>>> :
>>>
>>> > @@ -0,0 +1,6 @@
>>> +declare const docs: {
>>>
>>> shouldn't this be in the dist folder?
>>> ------------------------------
>>>
>>> In src/HubspotProvider-doc.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912914601>
>>> :
>>>
>>> > @@ -0,0 +1,12 @@
>>> +const docs = {
>>> +	get_info: {
>>>
>>> indentation should be 2 spaces
>>> ------------------------------
>>>
>>> In src/hubspot-provider.d.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912915456>
>>> :
>>>
>>> > @@ -0,0 +1,6 @@
>>> +declare function HubspotProvider(this: any, options: HubspotProviderOptions): {
>>>
>>> also, should this not be in dist?
>>> or did you define this manually?
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917054>
>>> :
>>>
>>> > @@ -0,0 +1,101 @@
>>> +const Pkg = require('../package.json');
>>> +
>>> +const Hubspot = ***@***.***/api-client')
>>> +
>>> +
>>> +type HubspotProviderOptions = {}
>>> +
>>> +function HubspotProvider(this: any, options : HubspotProviderOptions) {
>>> +  const seneca:any = this
>>> +
>>> +  const entityBuilder = this.export('provider/entityBuilder')
>>> +
>>> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
>>> +  async function get_info(this: any, _msg: any) {
>>>
>>> add a line to separate here
>>> if more messages are added later, we want all the messages to be together
>>> see:
>>> https://github.com/mgp/book-notes/blob/master/the-non-designers-design-book.markdown
>>> - proximity
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917262>
>>> :
>>>
>>> > +
>>> +  const entityBuilder = this.export('provider/entityBuilder')
>>> +
>>> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
>>> +  async function get_info(this: any, _msg: any) {
>>> +    return {
>>> +      ok: true,
>>> +      name: 'hubspot',
>>> +      version: Pkg.version,
>>> +      sdk: {
>>> +        name: 'hubspot',
>>> +	version: Pkg.dependencies['hubspot'],
>>> +      }
>>> +    }
>>> +  }
>>> + entityBuilder(this, {
>>>
>>> same here - line separation needed
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912918048>
>>> :
>>>
>>> > +      sdk: {
>>> +        name: 'hubspot',
>>> +	version: Pkg.dependencies['hubspot'],
>>> +      }
>>> +    }
>>> +  }
>>> + entityBuilder(this, {
>>> +   provider: {
>>> +     name: 'hubspot'
>>> +   },
>>> +   entity: {
>>> +     company: {
>>> +       cmd: {
>>> +         list: {
>>> +	     action: async function(this: any, entize: any, msg: any) {
>>> +	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
>>>
>>> use limit$
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912918761>
>>> :
>>>
>>> > +     company: {
>>> +       cmd: {
>>> +         list: {
>>> +	     action: async function(this: any, entize: any, msg: any) {
>>> +	       const limit = msg.q.limit || 10 // The maximum number of results to display per page
>>> +	       let res = (await this.shared.sdk.crm.companies.basicApi.getPage(limit)).results
>>> +	       let list = res.map((data) => entize(data))
>>> +	       return list
>>> +	     }
>>> +	 },
>>> +
>>> +	 load: {
>>> +	   action: async function(this: any, entize: any, msg: any) {
>>> +	     let idParts = msg.q.fields$
>>> +	     let id = msg.q.id
>>> +	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
>>>
>>> this line is too long - aim for max 80 chars (makes things like diff
>>> comparison easier)
>>> place the comment above the line
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912919528>
>>> :
>>>
>>> > +	 load: {
>>> +	   action: async function(this: any, entize: any, msg: any) {
>>> +	     let idParts = msg.q.fields$
>>> +	     let id = msg.q.id
>>> +	     let obj = await this.shared.sdk.crm.companies.basicApi.getById(id, idParts) // docs for the usage: https://developers.hubspot.com/docs/api/crm/companies
>>> +	     return entize(obj)
>>> +	   }
>>> +	 },
>>> +
>>> +	 save: {
>>> +	   action: async function(this: any, entize: any, msg: any) {
>>> +	     let ent = msg.ent
>>> +	     let id = ent.id
>>> +	     let desc = ent.properties.description
>>> +	     let obj
>>> +	     try{
>>>
>>> missing a space - just run https://prettier.io and all these formatting
>>> problems go away!
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912920034>
>>> :
>>>
>>> > +	       obj = await this.shared.sdk.crm.companies.basicApi.update(id, {properties: {description: desc}})
>>> +	     }
>>> +	     catch(err){
>>> +	       if(err.code >= 400 && err.code < 500){
>>> +		 return null
>>> +	       }
>>> +	       throw err
>>> +	     }
>>> +	     return entize(obj)
>>> +	   }
>>> +	 }
>>> +       }
>>> +     }
>>> +   }
>>> + })
>>> +
>>>
>>> too much space
>>>
>>> in general, use 2 lines for "major" contextual differences
>>> ------------------------------
>>>
>>> In src/hubspot-provider.ts
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#discussion_r912917701>
>>> :
>>>
>>> > +
>>> +  const entityBuilder = this.export('provider/entityBuilder')
>>> +
>>> +  seneca.message('sys:provider,provider:hubspot,get:info', get_info)
>>> +  async function get_info(this: any, _msg: any) {
>>> +    return {
>>> +      ok: true,
>>> +      name: 'hubspot',
>>> +      version: Pkg.version,
>>> +      sdk: {
>>> +        name: 'hubspot',
>>> +	version: Pkg.dependencies['hubspot'],
>>> +      }
>>> +    }
>>> +  }
>>> + entityBuilder(this, {
>>>
>>> also, the indentation is incorrect
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/senecajs/seneca-hubspot-provider/pull/1#pullrequestreview-1027424817>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AJILHFSNLQOAGO4KLYCORM3VSLFB7ANCNFSM5Z4TKO5Q>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***>
>>>
>>
